src/python/deepgnn/graph_engine/snark/decoders.py

```diff
@@ -34,6 +34,7 @@ def decode(self, line: str) -> Iterator[Tuple[int, int, int, float, list]]:
         """Decode the line of text.
 
         This is a generator that yields a node then its outgoing edges in order.
+        A node and its outgoing edges can be on different partitions.
```

Comment:
This is a non-trivial feature. Could you add an update documentation(the one in the doc folder) on how/when to use it?

Reply:
I dont think there is any case where someone would go out of their way to use this, but it is important for people to know so they dont avoid it. So I think it is best that we document it on the decoder and BinaryWriter docstings?